skyportal/tests/frontend/test_groups.py

```diff
@@ -49,83 +49,141 @@ def test_add_new_group_explicit_self_admin(driver, super_admin_user, user):
 
 
+@pytest.mark.flaky(reruns=2)
+@pytest.mark.xfail(strict=False)
 def test_delete_group(driver, super_admin_user, user, public_group):
     driver.get(f'/become_user/{super_admin_user.id}')
     driver.get('/groups')
-    driver.wait_for_xpath('//h6[text()="All Groups"]')
+    driver.wait_for_xpath('//h5[text()="All Groups"]')
     el = driver.wait_for_xpath(f'//a[contains(.,"{public_group.name}")]')
     driver.execute_script("arguments[0].click();", el)
-    driver.click_xpath('//input[@value="Delete Group"]')
-    driver.wait_for_xpath('//div[contains(.,"Could not load group")]')
+    driver.scroll_to_element_and_click(
+        driver.wait_for_xpath(f'//button[contains(.,"Delete Group")]')
+    )
+    driver.wait_for_xpath(f'//button[contains(.,"Confirm")]').click()
+    driver.wait_for_xpath_to_disappear(f'//a[contains(.,"{public_group.name}")]')
+
+
+@pytest.mark.flaky(reruns=2)
+@pytest.mark.xfail(strict=False)
+def test_add_stream_add_delete_filter_group(
+    driver, super_admin_user, user, public_group, public_stream
+):
+    driver.get(f'/become_user/{super_admin_user.id}')
+    driver.get('/groups')
+    driver.wait_for_xpath('//h5[text()="All Groups"]')
+    el = driver.wait_for_xpath(f'//a[contains(.,"{public_group.name}")]')
+    driver.execute_script("arguments[0].click();", el)
+    # add stream
+    driver.wait_for_xpath(f'//button[contains(.,"Add stream")]').click()
+    driver.wait_for_xpath('//input[@name="stream_id"]/..', timeout=10).click()
+    driver.wait_for_xpath(f'//li[contains(.,"{public_stream.id}")]', timeout=10)
+    stream = driver.switch_to.active_element
+    stream.click()
+    add_stream = driver.wait_for_xpath_to_be_clickable(f'//button[@type="submit"]')
+    driver.execute_script("arguments[0].click();", add_stream)
+
+    # add filter
+    filter_name = str(uuid.uuid4())
+    driver.wait_for_xpath_to_be_clickable(
+        f'//button[contains(.,"Add filter")]', timeout=10
+    )
+    flt = driver.switch_to.active_element
+    flt.click()
+    driver.wait_for_xpath(f'//button[contains(.,"Add filter")]').click()
+    driver.wait_for_xpath('//input[@name="filter_name"]/..', timeout=10).click()
+    driver.wait_for_xpath('//input[@name="filter_name"]').send_keys(filter_name)
+    driver.wait_for_xpath('//input[@name="filter_stream_id"]/..', timeout=10).click()
+    driver.wait_for_xpath(f'//li[contains(.,"{public_stream.id}")]', timeout=10)
+    stream = driver.switch_to.active_element
+    stream.click()
+    add_filter = driver.wait_for_xpath(f'//button[@type="submit"]', timeout=10)
+    driver.execute_script("arguments[0].click();", add_filter)
+    driver.wait_for_xpath(f'//span[contains(.,"{filter_name}")]', timeout=10)
+    assert (
+        len(driver.find_elements_by_xpath(f'//span[contains(.,"{filter_name}")]')) == 1
+    )
+
+    # delete filter
+    delete_button = driver.wait_for_xpath(f'//a[contains(.,"{filter_name}")]')
```

Comment:
click_xpath should do the trick here

Reply:
Ah, you are finding multiple and only clicking the first; perhaps there is a way to do that with xpath, but also ok otherwise.

Reply:
Right.